File: code/fe/types/event.ts
export interface EventItem {
  eventId: string;
  title: string;
  body: string;
  startTime: string; // "09:00:00"
  finishTime: string; // "11:30:00"
  eventDate: string; // "2026-04-02"
  schoolYear: number;
  term: number;
}

export interface EventListResponse {
  items: EventItem[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface GetEventsParams {
  Title?: string;
  SchoolYear?: number;
  Term?: number;
}

export interface CreateEventPayload {
  title: string;
  body: string;
  startTime: string;
  finishTime: string;
  eventDate: string;
  schoolYear: number;
  term: number;
}

File: code/be/School_Management.API/School_Management.API/Models/Domain/Event.cs
using System.ComponentModel.DataAnnotations;

namespace School_Management.API.Models.Domain
{
    public class Event
    {
        [Key]
        public Guid Id { get; set; }
        public string? Title { get; set; }
        public string? Body { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan FinishTime { get; set; }
        public DateOnly EventDate { get; set; }
        public int SchoolYear { get; set; }
        public int Term { get; set; }
    }
}
